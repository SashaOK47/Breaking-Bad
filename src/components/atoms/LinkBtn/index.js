import React from "react";
import style from "./styles.module.scss";
import { Link } from "react-router-dom";

const LinkBtn = (props) => {
  return (
    <Link className={style.linkBtn} {...props}>
      Назад в Каталог
    </Link>
  );
};
export default LinkBtn;
