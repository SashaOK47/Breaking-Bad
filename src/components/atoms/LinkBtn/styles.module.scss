@import "../../../assets/styles/main.scss";
.linkBtn {
  position: relative;
  display: inline-block;
  margin-bottom: 3rem;
  padding-left: 2rem;
  @include mainFont;
  @include tabletLg {
    margin-bottom: 1.5rem;
  }
  &::before {
    content: "";
    position: absolute;
    left: 0;
    top: 0.125rem;
    width: 1.5rem;
    height: 1.5rem;
    background: url(../../../assets/icons/arrow.svg) no-repeat;
    animation: arrow 0.8s infinite;
  }
  &::after {
    content: "";
    position: absolute;
    width: 0;
    bottom: -0.125rem;
    right: 0;
    height: 1px;
    background-color: $mainFontColor;
    transition: width 0.5s;
  }
  &:hover::after {
    width: calc(100% - 2rem);
  }
  &:focus-visible {
    @include boxShadowFocus;
  }
}
@keyframes arrow {
  0% {
    left: 0;
  }
  50% {
    left: -0.4rem;
  }
  100% {
    left: 0;
  }
}
