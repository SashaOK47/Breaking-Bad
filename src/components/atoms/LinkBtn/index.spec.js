import React from "react";
import LinkBtn from "./index";

const setUp = (props) => shallow(<LinkBtn {...props} />);

describe("Error component", () => {
  it("should render LinkBtn component", () => {
    const component = setUp({ location: "/characters" });
    expect(component).toMatchSnapshot();
  });
});
