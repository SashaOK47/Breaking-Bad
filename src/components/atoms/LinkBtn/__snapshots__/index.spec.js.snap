// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render LinkBtn component 1`] = `
<Link
  className="linkBtn"
  location="/characters"
>
  Назад в Каталог
</Link>
`;
